src/SConscript

```diff
@@ -42,9 +42,9 @@ for environ in [test_env, client_env, env]:
             environ.Append(CPPDEFINES = "LOCALEDIR='\"$localedirname\"'")
             if not os.path.isabs(env['localedirname']):
                 environ.Append(CPPDEFINES = "HAS_RELATIVE_LOCALEDIR")
-        if env['version_suffix'] and not env['prefsdir']:
-            environ['prefsdir'] = ".wesnoth$version_suffix"
         if environ['prefsdir']:
+            if env['version_suffix']:
+                environ['prefsdir'] += env['version_suffix']
```

Comment:
Instead of changing how this works it should set prefsdir to `~/.wesnoth$version_suffix`.

Reply:
I assume that wouldn't work on Windows?

Reply:
If we support windows here then that should also keep doing what it does now. So it looks like replace `~`with the user's documents folder.

Reply:
I'd prefer that the person building wesnoth just says where they want the userdata folder to be.

Reply:
Well, then why mess with `prefsdir` at all? You can already set `prefsdir` and that is respected currently. This just changes all the logic around. Why is that desirable?

If the goal is to not have `version_suffix` provide a default for `prefsdir` then that is all that should be done. I still find that of dubious value.

The current, documented behavior seems quite reasonable. `suffix that will be added to default values of prefsdir, program_suffix and datadirname`

An easy way to get mostly independent builds for example for the stable and dev version. So you can use them concurrently.

Reply:
Because getting the Windows Documents folder location seems to be non-trivial and I don't want to have to add that to scons.

Reply:
Why scons? wesnoth already does it. I just want to keep current functionality.

Reply:
I'm confused. These are scons options?

Reply:
But interpreting a relative path (that does not start with . or ..) as relative to HOME etc is done by wesnoth. As such interpreting ~ instead as meaning HOME etc should be done by wesnoth as well and is already done for macos and linux. Windows might as well do the same and then scons can default to ~/... instead and it works the same as before.

Reply:
So you're saying that on Windows, wesnoth should interpret `~` as pointing to My Games?

Reply:
My Games under the user's documents folder, yes. Same behavior as now but explicitly requesting a path in the user's home dir which this basically is on windows.

Reply:
Updated.